MekHQ/src/mekhq/gui/adapter/UnitTableMouseAdapter.java

```diff
@@ -605,454 +629,508 @@ public void mouseReleased(MouseEvent e) {
-
-            if (oneSelected) {
-                menuItem = new JMenuItem("Name Unit");
-                menuItem.setActionCommand("FLUFF_NAME");
-                menuItem.addActionListener(this);
-                menuItem.setEnabled(true);
-                popup.add(menuItem);
-            }
-            // sell unit
-            if (gui.getCampaign().getCampaignOptions().canSellUnits()) {
-                popup.addSeparator();
-                menuItem = new JMenuItem("Sell Unit");
-                menuItem.setActionCommand("SELL");
-                menuItem.addActionListener(this);
-                menuItem.setEnabled(!unit.isDeployed());
-                popup.add(menuItem);
-            }
-            // GM mode
-            menu = new JMenu("GM Mode");
-            menuItem = new JMenuItem("Remove Unit");
-            menuItem.setActionCommand("REMOVE");
-            menuItem.addActionListener(this);
-            menuItem.setEnabled(gui.getCampaign().isGM());
-            menu.add(menuItem);
-            menuItem = new JMenuItem("Strip Unit");
-            menuItem.setActionCommand("STRIP_UNIT");
-            menuItem.addActionListener(this);
-            menuItem.setEnabled(gui.getCampaign().isGM());
-            menu.add(menuItem);
-            if (oneSelected) {
-                menuItem = new JMenuItem(unit.isMothballed() ? "Activate Unit" : "Mothball Unit");
-                menuItem.setActionCommand(unit.isMothballed() ? COMMAND_GM_ACTIVATE : COMMAND_GM_MOTHBALL);
-                menuItem.addActionListener(this);
-                menuItem.setEnabled(gui.getCampaign().isGM());
-                menu.add(menuItem);
-            }
-            menuItem = new JMenuItem("Undeploy Unit");
-            menuItem.setActionCommand("UNDEPLOY");
-            menuItem.addActionListener(this);
-            menuItem.setEnabled(gui.getCampaign().isGM() && unit.isDeployed());
-            menu.add(menuItem);
-            if (unit.getCrew().size() < unit.getFullCrewSize()) {
-                menuItem = new JMenuItem("Add full complement");
-                menuItem.setActionCommand("HIRE_FULL_GM");
-                menuItem.addActionListener(this);
-                menuItem.setEnabled(unit.isAvailable() && gui.getCampaign().isGM());
-                menu.add(menuItem);
+                if (oneSelected) {
+                    menuItem = new JMenuItem("Name Unit");
+                    menuItem.setActionCommand(COMMAND_FLUFF_NAME);
+                    menuItem.addActionListener(this);
+                    popup.add(menuItem);
+                }
+
+                // sell unit
+                if (gui.getCampaign().getCampaignOptions().canSellUnits()) {
+                    popup.addSeparator();
+                    menuItem = new JMenuItem("Sell Unit");
+                    menuItem.setActionCommand(COMMAND_SELL);
+                    menuItem.addActionListener(this);
+                    menuItem.setEnabled(!unit.isDeployed());
```

Comment:
Should this be protected by `oneSelected`? It currently checks if the unit is deployed.

Reply:
This should be hidden behind a check that ensures that one is not deployed, instead of enabling/disabling if the first unit is deployed. I've fixed that